apps/OpenSpace/ext/launcher/include/sgctedit/windowcontrol.h

```diff
@@ -0,0 +1,176 @@
+ * Copyright (c) 2014-2022                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#ifndef __OPENSPACE_UI_LAUNCHER___WINDOWCONTROL___H__
+#define __OPENSPACE_UI_LAUNCHER___WINDOWCONTROL___H__
+
+#include <QCheckBox>
+#include <QColor>
+#include <QComboBox>
+#include <QFrame>
+#include <QIntValidator>
+#include <QLabel>
+#include <QLayout>
+#include <QLineEdit>
+#include <QPainter>
+#include <QPainterPath>
+#include <QPoint>
+#include <QPushButton>
+#include <QStandardItemModel>
+#include <QVector>
+#include <QWidget>
+
+#include <vector>
+#include <iostream>
+#include <sgct/config.h>
+
+
+class WindowControl : public QWidget
+{
+Q_OBJECT
+public:
+    explicit WindowControl(unsigned int nMonitors, unsigned int monitorIndex,
+        unsigned int windowIndex, std::vector<QRect>& monitorDims,
+        const QString* winColors, QWidget *parent = nullptr);
+    ~WindowControl();
+    void setDimensions(const QRectF& dimensions);
+    void setWindowChangeCallback(std::function<void(int, int, const QRectF&)> cb);
+    void setWebGuiChangeCallback(std::function<void(unsigned int)> cb);
+    void showWindowLabel(const bool show);
+    QVBoxLayout* initializeLayout();
+    QRectF& dimensions();
+    QLineEdit* lineEditSizeWidth();
```

Comment:
None of these functions returning the internal state are used anywhere and can be removed